tests/snuba/api/endpoints/test_organization_events_mep.py

```diff
@@ -3207,6 +3207,60 @@ def test_transaction_user_misery(self) -> None:
             },
         }
 
+    def _setup_user_misery(self, spec: OnDemandMetricSpec) -> None:
+        events = [
+            ("one", 300),
+            ("two", 300),
+            ("one", 3000),
+            ("two", 3000),
+            ("three", 400),
+            ("four", 4000),
+        ]
+        for index, event in enumerate(events):
+            email = f"{event[0]}@example.com"
+            tags = {"user.email": email}
+            if event[1] > 300:
+                tags["satisfaction"] = "frustrated"
+            self.store_on_demand_metric(
+                index,
+                spec=spec,
+                additional_tags=tags,
+                timestamp=self.day_ago + timedelta(hours=index),
+            )
+
+    def test_on_demand_user_misery_big_number(self):
+        field = "user_misery(300)"
+        query = ""
+        spec = OnDemandMetricSpec(
+            field=field,
+            query=query,
+            spec_type=MetricSpecType.DYNAMIC_QUERY,
+        )
+        self._setup_user_misery(spec)
+
+        resp = self.do_request(
+            {
+                "field": [field],
+                "project": self.project.id,
+                "query": query,
+                "dataset": "metricsEnhanced",
+                "useOnDemandMetrics": "true",
+                "onDemandType": "dynamic_query",
+            },
+        )
+        assert resp.data == {
+            "data": [{field: user_misery_formula(miserable_users=3, unique_users=4)}],
```

Comment:
If I change this value to 6/6 it will pass: `user_misery_formula(miserable_users=6, unique_users=6)`

If I keep it as-is, I will see the following error:
```
SENTRY_SNUBA_INFO=1 pytest -s -vv tests/snuba/api/endpoints/test_organization_events_mep.py -k test_on_demand_user_misery_big_number
========================================================================================================== test session starts ===========================================================================================================
platform darwin -- Python 3.11.6, pytest-8.0.0, pluggy-1.4.0 -- /Users/armenzg/code/sentry/.venv/bin/python3
cachedir: .pytest_cache
django: version: 5.0.2
rootdir: /Users/armenzg/code/sentry
configfile: pyproject.toml
plugins: fail-slow-0.3.0, rerunfailures-11.0, sentry-0.1.11, time-machine-2.13.0, xdist-3.0.2, anyio-3.7.1, django-4.8.0, cov-4.0.0
collected 145 items / 144 deselected / 1 selected

tests/snuba/api/endpoints/test_organization_events_mep.py::OrganizationEventsMetricsEnhancedPerformanceEndpointTestWithOnDemandMetrics::test_on_demand_user_misery_big_number Creating test database for alias 'default' ('test_region')...
Creating test database for alias 'control' ('test_control')...
api.metrics.totals.body:
 {'app_id': 'default',
 'consistent': False,
 'dataset': 'metrics',
 'parent_api': '<missing>',
 'query': 'MATCH (generic_metrics_sets) SELECT divide(plus(uniqIf(value, '
          'and(equals(metric_id, 9223372036854775938), '
          "equals(tags_raw[9223372036854776026], 'frustrated'))), 5.8875), "
          'plus(uniqIf(value, equals(metric_id, 9223372036854775938)), '
          '117.75)) AS `user_misery_300` WHERE org_id = 4553488812670976 AND '
          'project_id IN array(4553488812736513) AND timestamp >= '
          "toDateTime('2023-11-15T00:00:00') AND timestamp < "
          "toDateTime('2024-02-14T00:00:00') AND tags_raw[9223372036854776069] "
          "= '8cf1ca3d' AND metric_id IN array(9223372036854775938) LIMIT 101 "
          'OFFSET 0 GRANULARITY 86400',
 'tenant_ids': {'organization_id': 4553488812670976,
                'referrer': 'api.metrics.totals',
                'use_case_id': 'transactions'}}
api.metrics.totals.sql:
 SELECT (divide(plus(uniqCombined64MergeIf((value AS _snuba_value), equals((metric_id AS _snuba_metric_id), 9223372036854775938) AND equals((arrayElement(tags.raw_value, indexOf(tags.key, 9223372036854776026)) AS `_snuba_tags_raw[9223372036854776026]`), 'frustrated')), 5.8875), plus(uniqCombined64MergeIf(_snuba_value, equals(_snuba_metric_id, 9223372036854775938)), 117.75)) AS _snuba_user_misery_300)
  FROM generic_metric_sets_local
 WHERE equals(granularity, 3)
   AND equals((org_id AS _snuba_org_id), 4553488812670976)
   AND in((project_id AS _snuba_project_id), [4553488812736513])
   AND greaterOrEquals((timestamp AS _snuba_timestamp), toDateTime('2023-11-15T00:00:00', 'Universal'))
   AND less(_snuba_timestamp, toDateTime('2024-02-14T00:00:00', 'Universal'))
   AND equals((arrayElement(tags.raw_value, indexOf(tags.key, 9223372036854776069)) AS `_snuba_tags_raw[9223372036854776069]`), '8cf1ca3d')
   AND in(_snuba_metric_id, [9223372036854775938])
 LIMIT 101
OFFSET 0
FAILEDDestroying test database for alias 'default' ('test_region')...
Destroying test database for alias 'control' ('test_control')...


================================================================================================================ FAILURES ================================================================================================================
___________________________________________________________ OrganizationEventsMetricsEnhancedPerformanceEndpointTestWithOnDemandMetrics.test_on_demand_user_misery_big_number ____________________________________________________________
tests/snuba/api/endpoints/test_organization_events_mep.py:3252: in test_on_demand_user_misery_big_number
    assert resp.data == {
E   AssertionError: assert {'data': [{'user_misery(300)': 0.09606060606060605}], 'meta': {'fields': {'user_misery(300)': 'number'}, 'units': {'user_misery(300)': None}, 'isMetricsData': True, 'isMetricsExtractedData': True, 'tips': {}, 'datasetReason': 'unchanged', 'dataset': 'metricsEnhanced'}} == {'data': [{'user_misery(300)': 0.07299794661190964}], 'meta': {'fields': {'user_misery(300)': 'number'}, 'units': {'user_misery(300)': None}, 'isMetricsData': True, 'isMetricsExtractedData': True, 'tips': {}, 'datasetReason': 'unchanged', 'dataset': 'metricsEnhanced'}}
E
E     Common items:
E     {'meta': {'dataset': 'metricsEnhanced',
E               'datasetReason': 'unchanged',
E               'fields': {'user_misery(300)': 'number'},
E               'isMetricsData': True,
E               'isMetricsExtractedData': True,
E               'tips': {},
E               'units': {'user_misery(300)': None}}}
E     Differing items:
E     {'data': [{'user_misery(300)': 0.09606060606060605}]} != {'data': [{'user_misery(300)': 0.07299794661190964}]}
E
E     Full diff:
E       {
E           'data': [
E               {
E     -             'user_misery(300)': 0.07299794661190964,
E     +             'user_misery(300)': 0.09606060606060605,
E               },
E           ],
E           'meta': {
E               'dataset': 'metricsEnhanced',
E               'datasetReason': 'unchanged',
E               'fields': {
E                   'user_misery(300)': 'number',
E               },
E               'isMetricsData': True,
E               'isMetricsExtractedData': True,
E               'tips': {},
E               'units': {
E                   'user_misery(300)': None,
E               },
E           },
E       }
======================================================================================================== short test summary info =========================================================================================================
FAILED tests/snuba/api/endpoints/test_organization_events_mep.py::OrganizationEventsMetricsEnhancedPerformanceEndpointTestWithOnDemandMetrics::test_on_demand_user_misery_big_number - AssertionError: assert {'data': [{'user_misery(300)': 0.09606060606060605}], 'meta': {'fields': {'user_misery(300)': 'number'}, 'units': {'user_misery(300)': None}, 'isMetricsData': True, 'isMetricsExtractedData': True, 'tips': {...
=================================================================================================== 1 failed, 144 deselected in 4.37s ====================================================================================================
```